sympy/matrices/matrices.py

```diff
@@ -4029,6 +4032,8 @@ def solve(self, rhs, method='GE'):
             elif self.rows > self.cols:
                 raise ValueError('For over-determined system, M, having '
                                  'more rows than columns, try M.solve_least_squares(rhs).')
+        if method == 'GE':
```

Comment:
```
if method == 'GJ':
    soln, param = self.gauss_jordan_solve(rhs)
    if param:
        raise ValueError ( as given in above comment )
    return soln
else:
```

Reply:
This would then need a test like
```
A = Matrix([[1, 2], [2, 4]])
b = Matrix([[ 3], [-3]])
raises(ValueError, lambda: A.solve(b))
A = Matrix([[1, 2], [2, 4]])
b = Matrix([[ 3], [6]])
raises(ValueError, lambda: A.solve(b))
```
to confirm that even when GJ is used, ValueErrors are raised for non-invertible systems (which is what the current solve implicitly does).

Reply:
Thanks @smichr for the review . I agree with your changes . Should i commit the change or wait for others approval as well .

Reply:
I recommend commiting the changes. This is the only way to preserve the current behavior of the function. Because it is currently based on `inv` it raises an error when the matrix is singular. G-J won't do that so the changes I recommend catch that case and raise an error while still using a faster routine to calculate the solution.